src/event/player/PlayerResourcePackOfferEvent.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+/*
+ *
+ *  ____            _        _   __  __ _                  __  __ ____
+ * |  _ \ ___   ___| | _____| |_|  \/  (_)_ __   ___      |  \/  |  _ \
+ * | |_) / _ \ / __| |/ / _ \ __| |\/| | | '_ \ / _ \_____| |\/| | |_) |
+ * |  __/ (_) | (__|   <  __/ |_| |  | | | | | |  __/_____| |  | |  __/
+ * |_|   \___/ \___|_|\_\___|\__|_|  |_|_|_| |_|\___|     |_|  |_|_|
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * @author PocketMine Team
+ * @link http://www.pocketmine.net/
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace pocketmine\event\player;
+
+use pocketmine\event\Event;
+use pocketmine\player\PlayerInfo;
+use pocketmine\resourcepacks\ResourcePack;
+use function assert;
+
+/**
+ * Called after a player authenticates and is being offered resource packs to download.
+ *
+ * This event should be used to decide which resource packs to offer the player and whether to require the player to
+ * download the packs before they can join the server.
+ */
+class PlayerResourcePackOfferEvent extends Event{
+
+	private readonly PlayerInfo $playerInfo;
+
+	/**
+	 * @param ResourcePack[] $resourcePacks
+	 */
+	public function __construct(
+		?PlayerInfo $playerInfo,
+		private array $resourcePacks,
+		private bool $mustAccept
+	){
+		assert($playerInfo !== null, "PlayerInfo cannot be null");
```

Comment:
?????

Reply:
It resolves with phpstan this way. Is a new rule preferred?

Reply:
No, I said pass it in the constructor of the ResourcePacksPacketHandler. Not sure what you thought I meant ...

Reply:
I did that with [007a29c](https://github.com/pmmp/PocketMine-MP/pull/6249/commits/007a29cdca06070bb197b61e422f0568ee1e9ac3) but it causes phpstan analysis to fail so I moved it here

Reply:
All you did was move the assert. That's not what I said.